### PR TITLE
Fix clang 3.8 arm

### PR DESCRIPTION
**Pull request by julek-wolfssl**
When compiling wolfSSL using clang 3.8 on ARM with ASM optimizations enabled, clang complains about the sizes of variables not matching register sizes.
A snippet of the errors appearing:
```
wolfcrypt/src/port/arm/armv8-poly1305.c:181:27: error: value size does not match register size specified by the
      constraint and modifier [-Werror,-Wasm-operand-widths]
          [finished] "r" (ctx->finished)
                          ^
wolfcrypt/src/port/arm/armv8-poly1305.c:76:21: note: use constraint modifier "w"
        "CMP        %[finished], #0      \n\t"
                    ^~~~~~~~~~~
                    %w[finished]
wolfcrypt/src/port/arm/armv8-poly1305.c:181:27: error: value size does not match register size specified by the
      constraint and modifier [-Werror,-Wasm-operand-widths]
          [finished] "r" (ctx->finished)
                          ^
wolfcrypt/src/port/arm/armv8-poly1305.c:78:21: note: use constraint modifier "w"
        "CSET       %[finished], EQ      \n\t"
                    ^~~~~~~~~~~
                    %w[finished]
wolfcrypt/src/port/arm/armv8-poly1305.c:181:27: error: value size does not match register size specified by the
      constraint and modifier [-Werror,-Wasm-operand-widths]
          [finished] "r" (ctx->finished)
                          ^
wolfcrypt/src/port/arm/armv8-poly1305.c:80:21: note: use constraint modifier "w"
        "LSL        %[finished], %[finished], #24 \n\t"
                    ^~~~~~~~~~~
                    %w[finished]
wolfcrypt/src/port/arm/armv8-poly1305.c:181:27: error: value size does not match register size specified by the
      constraint and modifier [-Werror,-Wasm-operand-widths]
          [finished] "r" (ctx->finished)
                          ^
wolfcrypt/src/port/arm/armv8-poly1305.c:80:34: note: use constraint modifier "w"
        "LSL        %[finished], %[finished], #24 \n\t"
                                 ^~~~~~~~~~~
                                 %w[finished]
wolfcrypt/src/port/arm/armv8-poly1305.c:181:27: error: value size does not match register size specified by the
      constraint and modifier [-Werror,-Wasm-operand-widths]
          [finished] "r" (ctx->finished)
                          ^
wolfcrypt/src/port/arm/armv8-poly1305.c:104:26: note: use constraint modifier "w"
        "ORR        x17, %[finished], x17, LSR #40 \n\t"
                         ^~~~~~~~~~~
                         %w[finished]
wolfcrypt/src/port/arm/armv8-poly1305.c:865:27: error: value size does not match register size specified by the
      constraint and modifier [-Werror,-Wasm-operand-widths]
          [finished] "r" (ctx->finished)
                          ^
wolfcrypt/src/port/arm/armv8-poly1305.c:216:21: note: use constraint modifier "w"
        "CMP        %[finished], #0      \n\t"
                    ^~~~~~~~~~~
                    %w[finished]
  CC       wolfcrypt/src/src_libwolfssl_la-chacha20_poly1305.lo
```

### CODE REVIEW

File: wolfcrypt/src/port/arm/armv8-chacha.c [hunk 1]
```diff
@@ -1673,7 +1673,12 @@ static WC_INLINE int wc_Chacha_encrypt_128(const word32 input[CHACHA_CHUNK_WORDS
 }
 
 static WC_INLINE void wc_Chacha_encrypt_64(const word32* input, const byte* m,
-                                           byte* c, word32 bytes)
+                                           byte* c,
+#ifdef __aarch64__
+                                           word64 bytes)
```
Comment (SparkiDev) [inline wolfcrypt/src/port/arm/armv8-chacha.c]: 'bytes' is a 32-bit value.
Cast it like you did below.
Same above.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Comment by SparkiDev** [comment]
Squash the commits - one of the commits is by root@localhost.

**Comment by SparkiDev** [comment]
Code as checked in didn't pass tests with clang 3.5.
clang 3.5 didn't like cast when indicating bytes changes so I had to make it read-only in the asm variable list to make it work (moved it to second list of variables/registers).
 [bytes] "+r" ((word64)bytes) didn't work
 [bytes] "r" ((word64)bytes) did work

**Comment by julek-wolfssl** [comment]
bytes has to be placed in the output list because it is changed in wc_Chacha_encrypt_64 and wc_Chacha_encrypt_320. If it is placed in the read-only list the optimizer blindly assumes that the value doesn't change and this messes up the rest of the algorithm. In my last commit I kept the 32 bit version as the parameter but cast it to a 64 bit local variable in the 64 bit state.

**Review by SparkiDev — APPROVED** [review]

**Comment by SparkiDev** [comment]
Squash commits

**Pull request merged**